### PR TITLE
js_gallery_DOM: Cypress tests added

### PR DESCRIPTION
**Pull request by iBrianWarner**

### CODE REVIEW

File: cypress/integration/js_gallery.spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable strict */
+describe(('Tests'), () => {
+  before(() => {
+    cy.visit('/');
+  });
+  
+  it('A click can be on a small `img` image', () => {
+    cy.get('ul> li:nth-child(2)').click()
```
Comment (YuriiHoliuk) [inline cypress/integration/js_gallery.spec.js:8]:
```suggestion
    cy.get('ul > li:nth-child(2)').click()
```

File: cypress/integration/js_gallery.spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    cy.get('[id="largeImg"]').should('have.attr', 'src', 'http://localhost:8080/images/landscape-second.png')
```
Comment (YuriiHoliuk) [inline cypress/integration/js_gallery.spec.js:9]: do not repeat code, move URL to constant or retrieve from cypress.
use alias for `cy.get('[id="largeImg"]')`

File: cypress/integration/js_gallery.spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    cy.get('ul> li:nth-child(3)').click()
```
Comment (YuriiHoliuk) [inline cypress/integration/js_gallery.spec.js:11]: I think better to use classes

File: cypress/integration/js_gallery.spec.js (hunk 1 above)
Comment (iBrianWarner) [reply cypress/integration/js_gallery.spec.js]: виправив

File: cypress/integration/js_gallery.spec.js (hunk 2 above)
Comment (iBrianWarner) [reply cypress/integration/js_gallery.spec.js]: створив змінну url в describe
створив alias @largeImg в beforeEach

File: cypress/integration/js_gallery.spec.js (hunk 3 above)
Comment (iBrianWarner) [reply cypress/integration/js_gallery.spec.js]: замінив селектори в другому тесті на більш зручні (ящетаю), в першому не знайшов як моно оптимізувати - і так норм працює)

File: cypress/integration/galleryDom.spec.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+describe(('Gallery application'), () => {
```
Comment (roman-mate) [inline cypress/integration/galleryDom.spec.js:3]: Are parentheses around 'Gallery application' needed?

File: cypress/integration/galleryDom.spec.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  const url = 'http://localhost:8080';
+  const imageSec = '/images/landscape-second.png';
+  const imageThird = '/images/landscape-third.png';
+  const imageFourth = '/images/landscape-fourth.png';
+  const imageFifth = '/images/landscape-fifth.png';
+  const imageFirst = '/images/landscape-first.png';
+
+  beforeEach(() => {
+    cy.visit('/');
+    cy.get('[id="largeImg"]').as('largeImg');
```
Comment (roman-mate) [inline cypress/integration/galleryDom.spec.js:13]: #largeImg

File: cypress/integration/galleryDom.spec.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('@largeImg').should('have.attr', 'src', url + imageThird);
+
+    cy.get('ul li:nth-child(4)').click();
+    cy.get('@largeImg').should('have.attr', 'src', url + imageFourth);
+
+    cy.get('ul li:nth-child(5)').click();
+    cy.get('@largeImg').should('have.attr', 'src', url + imageFifth);
+
+    cy.get('ul li:nth-child(1)').click();
+    cy.get('@largeImg').should('have.attr', 'src', url + imageFirst);
```
Comment (roman-mate) [inline cypress/integration/galleryDom.spec.js:30]: can we do this in a loop?

File: cypress/integration/galleryDom.spec.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  const imageFifth = '/images/landscape-fifth.png';
+  const imageFirst = '/images/landscape-first.png';
+
+  beforeEach(() => {
+    cy.visit('/');
+    cy.get('[id="largeImg"]').as('largeImg');
+  });
+
+  it('can click on a small `img` image', () => {
+    cy.get('ul li:nth-child(2)').click();
```
Comment (roman-mate) [inline cypress/integration/galleryDom.spec.js:17]: is there a better selector then "ul li:nth-child(n)"?

File: cypress/integration/galleryDom.spec.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+    cy.get('ul li:nth-child(5)').click();
+    cy.get('@largeImg').should('have.attr', 'src', url + imageFifth);
+
+    cy.get('ul li:nth-child(1)').click();
+    cy.get('@largeImg').should('have.attr', 'src', url + imageFirst);
+  });
+
+  it('can click on `a` outside of image', () => {
+    cy.get('[title="Image 2"]').click();
```
Comment (roman-mate) [inline cypress/integration/galleryDom.spec.js:34]: Is there a better selector?

**Review by roman-mate — CHANGES_REQUESTED** [review]

File: cypress/integration/galleryDom.spec.js (hunk 6 above)
Comment (iBrianWarner) [reply cypress/integration/galleryDom.spec.js]: done

File: cypress/integration/galleryDom.spec.js (hunk 7 above)
Comment (iBrianWarner) [reply cypress/integration/galleryDom.spec.js]: all li elements are equal, li:nth-child is shortest

File: cypress/integration/galleryDom.spec.js (hunk 8 above)
Comment (iBrianWarner) [reply cypress/integration/galleryDom.spec.js]: this is a unique parameter, another way I see is 'li:nth-child a'

File: cypress/integration/galleryDom.spec.js (hunk 4 above)
Comment (iBrianWarner) [reply cypress/integration/galleryDom.spec.js]: nope, deleted

File: cypress/integration/galleryDom.spec.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        cy.get(`${selector}(${i + 1})`).click();
+        cy.get('@largeImg').should('have.attr', 'src', `${url}${images[i]}`);
+      };
+    });
+
+    Cypress.Commands.add('abeenSelected', (selector) => {
+      for (let i = 0; i < images.length; i++) {
+        cy.get(`[${selector} ${i + 1}"]`).click();
+        cy.get('@largeImg').should('have.attr', 'src', `${url}${images[i]}`);
+      };
```
[the remark this answers is not in the document]
Comment (iBrianWarner) [reply cypress/integration/galleryDom.spec.js]: I've realized it was a mistake to create a command for only one test and deleted all commands

**Review by brainwashers — APPROVED** [review]

File: cypress/integration/galleryDom.spec.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const url = 'http://localhost:8080';
```
Comment (roman-mate) [inline cypress/integration/galleryDom.spec.js:3]: this should be replacable with Cypress.config().baseUrl.

**Review by roman-mate — APPROVED** [review]
Awesome, I like it. Just a final nit. I my suggestion doesn't work - you can merge.

File: cypress/integration/galleryDom.spec.js (hunk 10 above)
Comment (iBrianWarner) [reply cypress/integration/galleryDom.spec.js]: yes, it is. replaced to Cypress.config().baseUrl

**Review by roman-mate — APPROVED** [review]

**Pull request merged**